### PR TITLE
Fix warnings deprecation and test errors

### PR DESCRIPTION
**Pull request by alexzerah**
## Details
- Fix warnings deprecation

## Check
- Tests are ok 👍
- CS are ok 👍

### CODE REVIEW

**Comment by lyrixx** [comment]
IMHO, we should add a conflict rule in composer.json to forgot version of symfony that are not compatible anymore. This changes landed in symfony 4.2. So now this bundle is incompatible with symfony 3.4 :grimacing: 

Another way to deal with it, it to support both methods: https://github.com/symfony/monolog-bundle/blob/635bb23451665247ad676f6b657d3f8c52f94705/DependencyInjection/Configuration.php#L338-L339

**Comment by alexzerah** [comment]
> IMHO, we should add a conflict rule in composer.json to forgot version of symfony that are not compatible anymore. This changes landed in symfony 4.2. So now this bundle is incompatible with symfony 3.4 😬
> 
> Another way to deal with it, it to support both methods: https://github.com/symfony/monolog-bundle/blob/635bb23451665247ad676f6b657d3f8c52f94705/DependencyInjection/Configuration.php#L338-L339

Second method seems better for me. Is it better to do that ?

File: composer.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
         "symfony/dependency-injection": "~3.0|~4.0",
         "symfony/framework-bundle": "~3.0|~4.0",
         "symfony/http-kernel": "~3.0|~4.0",
-        "symfony/phpunit-bridge": "^3.3|~4.0",
+        "symfony/phpunit-bridge": "^4.0",
```
Comment (lyrixx) [inline composer.json]: I would event bump to 4.3

File: tests/Functional/Fixtures/symfony/app/config/config.yml [hunk 2]
```diff
@@ -8,24 +8,24 @@ services:
...
     secret: 'azertyuiop'
     router:
-        resource: '%kernel.root_dir%/config/routing.yml'
+        resource: '%kernel.project_dir%/app/config/routing.yml'
         strict_requirements: ~
     session:
         # http://symfony.com/doc/current/reference/configuration/framework.html#handler-id
         handler_id:  session.handler.native_file
-        save_path:   "%kernel.root_dir%/../var/sessions/%kernel.environment%"
+        save_path:   "%kernel.project_dir%/app/var/sessions/%kernel.environment%"
```
Comment (pyrech) [inline tests/Functional/Fixtures/symfony/app/config/config.yml]: app is wrong here
```suggestion
        save_path:   "%kernel.project_dir%/var/sessions/%kernel.environment%"
```

File: composer.json [hunk 3]
```diff
@@ -34,13 +34,15 @@
...
         "phpdocumentor/reflection-docblock": "^4.0",
         "phpspec/prophecy": "^1.7",
-        "symfony/config": "~3.0|~4.0",
-        "symfony/dependency-injection": "~3.0|~4.0",
-        "symfony/framework-bundle": "~3.0|~4.0",
-        "symfony/http-kernel": "~3.0|~4.0",
-        "symfony/phpunit-bridge": "^3.3|~4.0",
-        "symfony/twig-bundle": "~3.0|~4.0",
-        "symfony/yaml": "~3.0|~4.0"
+
```
Comment (lyrixx) [inline composer.json]: to remove

File: composer.json [hunk 4]
```diff
@@ -34,13 +34,15 @@
...
+        "symfony/config": "~3.4|~4.0",
```
Comment (lyrixx) [inline composer.json]: We need to adds `conflicts` rule for all theses deps

File: tests/Functional/Fixtures/symfony/app/config/config.yml [hunk 5]
```diff
@@ -8,24 +8,24 @@ services:
...
 
 doctrine:
     dbal:
         driver:   pdo_sqlite
-        path:     "%kernel.root_dir%/data/data.sqlite"
+        path:     "%kernel.project_dir%/app/data/data.sqlite"
```
Comment (lyrixx) [inline tests/Functional/Fixtures/symfony/app/config/config.yml]: `app` does not exists anymore in SF 4.X
Let's move this to ` "%kernel.project_dir%/data/data.sqlite"`

Comment (alexzerah) [reply tests/Functional/Fixtures/symfony/app/config/config.yml]: If we do that, do we have to update the folder structure for all the SF app (which is in an `app` folder)?

Comment (lyrixx) [reply tests/Functional/Fixtures/symfony/app/config/config.yml]: Could be better indeed, but could be left also. It's up to you

Comment (alexzerah) [reply tests/Functional/Fixtures/symfony/app/config/config.yml]: Because this is a big change, I created an issue to fix the old architecture -> #41

**Pull request merged**